tests/lib/assertions.py

```diff
@@ -211,7 +211,8 @@ def assert_error_with_optional_correlation_id(resp, code, error, correlation_id)
             else:
                 assert error in response_errors
 
-        assert resp.headers.get("X-Correlation-Id") == correlation_id
+        if correlation_id is not None:
```

Comment:
Does this need to be here? When we've passed `None` in as the correlation id in the past this hasn't caused us any issues

Reply:
Removed